qiskit/dagcircuit/dagnode.py

```diff
@@ -12,16 +12,146 @@
 
 # pylint: disable=redefined-builtin
 
-"""Object to represent the information at a node in the DAGCircuit."""
+"""Objects to represent the information at a node in the DAGCircuit."""
 
 import warnings
 
 from qiskit.exceptions import QiskitError
 
 
+class DAGNodeP:
+    """Parent class for OpNode, InNode, and OutNode used by DAGCircuit."""
+
+    __slots__ = ["_qargs", "cargs", "sort_key", "_node_id"]
+
+    def __init__(self, qargs=None, cargs=None, nid=-1):
+        """Create a node"""
+        self._qargs = qargs if qargs is not None else []
+        self.cargs = cargs if cargs is not None else []
+        self.sort_key = str(self._qargs)
```

Comment:
Are these specific to only `OpNode`s?

Reply:
No. The _qargs are used to create the sort_key which is used in `dagcircuit.topological_nodes` for all types of nodes. That's why they're handled here by the parent.